src/modules/VirtConfig.rb

```diff
@@ -317,6 +318,70 @@ def LXCDialog
         )
       end
 
+      Builtins.y2milestone("VirtConfig::ConfigureDom0: Checking for Installed Patterns and Packages")
+      is_xen_server_installed = false
```

Comment:
You actually do not need to initialize the variables, you can use this Ruby trick:

```ruby
if false
  is_xen_server_installed = true
end
# here `is_xen_server_installed` is `nil`
```
After the block here `is_xen_server_installed` is defined and has `nil` value although the line with the assignment was  not executed. (You can check this in `irb`.)

And `nil` will be evaluated as `false` in `if is_xen_server_installed` condition later (even after removing that `== true` comparison). That means you can simplify the code a quite a lot...